### ISSUE TITLE
CS8784 and CS8785 severity can not be changed to error via .editorconfig

### ISSUE THREAD

**Issue by pavel-mikula-sonarsource** [dotnet/roslyn#81479]
**Version Used**: Compiler version: '5.0.0-2.25529.7 (68a9207c)'. Language version: 14.0.

**Steps to Reproduce**:
1. Implement grumpy generator and use it in the project
```
[Generator]
public class GrumpyGenerator : IIncrementalGenerator
{
    public void Initialize(IncrementalGeneratorInitializationContext context) =>
        throw new Exception("No!");
}
```
2.  Set CS8784 and CS8785 severity to error in editor config
```
[*]
dotnet_diagnostic.CS8784.severity = error   
dotnet_diagnostic.CS8785.severity = error   
```
I tried also this, as the issue is reported on CSC file, but with no luck:
```
[CSC]
dotnet_diagnostic.CS8784.severity = error   
dotnet_diagnostic.CS8785.severity = error   
```

3. Rebuild the project

**Expected Behavior**:

Build should fail with CS8784 as error.

If a source generator fails, a significant prerequisite is missing from the build. I would even propose to change those two into Errors by default.

**Actual Behavior**:

Build contains CS8784 as warning.

This warning is hard to spot, as its hidden behind new unexpected compilation errors, as generated code is missing and project does not compile.

**Comment by jjonescz** [comment]
Looks like this works with `.globalconfig` and with `<WarningsAsErrors>`, only `.editorconfig` seems broken.

**Comment by jjonescz** [comment]
And that seems to be by design. As you noted, the error isn't attached to any source .cs file, hence `.editorconfig` rules don't apply to it. You should use `.globalconfig` instead. See also https://github.com/dotnet/roslyn/issues/47707.

**Comment by pavel-mikula-sonarsource** [comment]
Díky
